### DISCUSSION TITLE
Cannot be connected in the campus network

### DISCUSSION THREAD

**Discussion by edge-sky** [rustdesk/rustdesk#7339] [Q&A]
I built the server on the server of Alibaba Cloud and and set relay server in client.It is fine when my phone and computer used 4G network, but when I used my campus network, phone and computer are unable to connect remotely. 
Before I use rustdesk, I has tried like toDesk and teamviewer, they are not normal access, seems to be blacklisted by the upstream network equipment. But I use rustdesk with my relay server, why it still can't be used? Is the related net prot is ban?
![1709881851308](https://github.com/rustdesk/rustdesk/assets/95955401/748ac420-d496-4669-b374-0599081a647e)
<img width="623" alt="{A3A10FE9-C266-4d23-AC05-ED9B485ABC29}" src="https://github.com/rustdesk/rustdesk/assets/95955401/c255ef06-00ec-4b9c-9f5b-837410f33851">

**Comment by 21pages** [comment]
Could you try https://github.com/rustdesk/rustdesk/releases/tag/nightly on the control side? There should be a pop-up prompting you to connect the relay.

**Comment by 21pages** [comment]
> Yes, I have been installing this edition on my phone and I want to control my computer. But the pop-up still shows 'Connection reset by peer error 104' if I use my campus network. I think the relay configuration is already set because it can't show my computer online without setting the ID and key

What if connect with "ID/r"? eg: 123456/r

**Comment by 21pages** [comment]
![1709885599050](https://github.com/rustdesk/rustdesk/assets/14891774/93d035c2-18eb-43d9-a78c-434bfa1c3439)

**Reply by edge-sky** [reply]
![1709885949534](https://github.com/rustdesk/rustdesk/assets/95955401/2a6ee2dd-e6cf-4212-8135-45c947b28e0a)
![1709885949521](https://github.com/rustdesk/rustdesk/assets/95955401/54083060-21ed-40dd-b604-ea7b8308c94c)

**Reply by 21pages** [reply]
What about select the `Always connect via relay` option 

![d77d832a408fbd22a3e478ad0a58c9f](https://github.com/rustdesk/rustdesk/assets/14891774/26a97284-7533-40f6-8f7a-3758c74b0fd0)

**Comment by edge-sky** [comment]
I try, it still can't connect to my computer with campus network. I think the network make some limits but I don't know how it do.

**Comment by edge-sky** [comment]
I&nbsp;did&nbsp;it,&nbsp;but&nbsp;it&nbsp;also&nbsp;show&nbsp;&quot;Connection&nbsp;reset&nbsp;by&nbsp;peer&nbsp;(os&nbsp;error&nbsp;104)&quot;


---Original---
From: ***@***.***&gt;
Date: Fri, Mar 8, 2024 16:56 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [rustdesk/rustdesk] Cannot be connected in the campus network(Discussion #7339)




 
What about select the Always connect via relay option
 
d77d832a408fbd22a3e478ad0a58c9f.jpg (view on web)
 
&mdash;

**Comment by xlionjuan** [comment]
Try tcp ping
https://play.google.com/store/apps/details?id=ua.com.streamsoft.pingtools

**Reply by xlionjuan** [reply]
Click ping submenu 
![Screenshot_20240311-105804.png](https://github.com/rustdesk/rustdesk/assets/25243894/30951261-83e2-41a5-9143-4a8a05cc8a9a)

Click settings and change mode to tcp, and enter the port
And change the hostname to your host.
![Screenshot_20240311-105835.png](https://github.com/rustdesk/rustdesk/assets/25243894/136dd8f5-d9ca-4e5a-a31e-bb29b4bc8d6e)

**Reply by xlionjuan** [reply]
Windows:
https://www.elifulkerson.com/projects/tcping.php

**Comment by edge-sky** [comment]
Thanks for your advice, I just see your reply today. The campus network is able to use remote software and TCP ping is pass now.

---Original---
From: ***@***.***&gt;
Date: Mon, Mar 11, 2024 11:02 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [rustdesk/rustdesk] Cannot be connected in the campus network(Discussion #7339)




 
Click ping submenu
 Screenshot_20240311-105804.png (view on web)
 
Click settings and change mode to tcp, and enter the port
 And change the hostname to your host.
 Screenshot_20240311-105835.png (view on web)

**Reply by xlionjuan** [reply]
So maybe all UDP are blocked?

**Comment by edge-sky** [comment]
I&nbsp;don't&nbsp;know,&nbsp;but&nbsp;search&nbsp;by&nbsp;bing,&nbsp;some&nbsp;other&nbsp;campus&nbsp;through&nbsp;ban&nbsp;the&nbsp;RDP&nbsp;protocol&nbsp;or&nbsp;some&nbsp;ports&nbsp;to&nbsp;prevent&nbsp;remote&nbsp;control.


---Original---
From: ***@***.***&gt;
Date: Fri, Mar 22, 2024 17:08 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [rustdesk/rustdesk] Cannot be connected in the campus network(Discussion #7339)




 
So maybe all UDP are blocked?
 
&mdash;
